### ISSUE TITLE
Async installation

### ISSUE THREAD

**Issue by xsh005** [nvim-neorocks/rocks.nvim#107]
Right now rocks.nvim would block on `Rocks sync` (but only while syncing rocks.nvim itself), it would block the ui. It would be nice if it's not the case.

**Comment by mrcjkb** [comment]
🤔 rocks sync should be async. Which, if I understand your issue correctly, is the case as long as rocks.nvim isn't updating itself?
If so, then the case in which it blocks is a bug.

**Comment by xsh005** [comment]
Yes, it unblocks after it has updated itself (at least according to the progress UI)

[Screencast from 2024-01-01 18-27-25.webm](https://github.com/nvim-neorocks/rocks.nvim/assets/90909185/d82364e5-dfc5-4bc1-bab1-54eb9ad13b3c)

**Comment by mrcjkb** [comment]
🤔 I think this may be a `rocks-git.nvim` issue.
